provider-example/src/lib.rs

```diff
@@ -6,6 +6,7 @@ mod hash;
 mod hmac;
 mod hpke;
 mod kx;
+pub mod sign;
```

Comment:
I'm not sure I see the reason this needs to be `pub`.
```suggestion
mod sign;
```